src/java/org/apache/cassandra/index/sai/disk/MemtableTermsIterator.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk;
```

Comment:
nit: Seems odd that `MemtableTermsIterator` is in the `disk` package?

Reply:
I agree. I've moved it to `sai.memory`. At the same time I have moved `TermsIterator` to `sai.utils`. This is not an ideal place for it but the only other place it could go would be `sai.memory` as well. It's only used by `MemtableTermsIterator` in code although the `LiterIndexSegmentTermsReader.TermsScanner` uses it but that is only used in tests.

One option would be to remove `TermsIterator` entirely. We would then use `MemtableTermsIterator` in code and `TermsScanner` in tests. I think I have previously said that my preference would be to try and get `TermsScanner` out of the code and into test.

I'll see what this change looks like.